### ISSUE TITLE
mkdir /var/lib/kubelet/pods/xxxxx/volumes/kubernetes.io~csi/pvc-xxx/mount: file exists

### ISSUE THREAD

**Issue by papanito** [kadalu/kadalu#121]
I try to install [`nextcloud` with `mariadb`](https://github.com/helm/charts/tree/master/stable/nextcloud) with helm, using `kadalu.replica1` as persistent storage. This fails with an error related to storage

```
pod/nextcloud-mariadb-master-0 
Error: failed to start container "mariadb": Error response from daemon: error while creating mount source path '/var/lib/kubelet/pods/733aabd1-ed37-42a6-beb2-3e29ee4c24d5/volumes/kubernetes.io~csi/pvc-e280b38a-edd4-495e-8bd5-09389bf58238/mount': 
mkdir /var/lib/kubelet/pods/733aabd1-ed37-42a6-beb2-3e29ee4c24d5/volumes/kubernetes.io~csi/pvc-e280b38a-edd4-495e-8bd5-09389bf58238/mount: file exists
```

**Steps to reproduce**

Install `nextcloud` using persistent volume claim when installing via helm chart

1. Create `values.yaml` defining persistence using `ReadWriteMany` (shows only relevant part)

   ```yaml
    nextcloud:
      host: nc.xxxx.dev
      username: xxxx
      password: REDACTED
      update: 0
      datadir: /var/www/html/data
      tableprefix:

    internalDatabase:
      enabled: false
      name: nextcloud

    externalDatabase:
      enabled: false

    mariadb:
      enabled: true

      securityContext:
        fsGroup: 82
        runAsUser: 33

      db:
        name: nextcloud
        user: nextcloud
        password: xxx

      persistence:
        enabled: false
        accessMode: ReadWriteMany
        size: 8Gi

      master:
        persistence:
          accessModes: 
            - ReadWriteMany
      slave:
        persistence:
          accessModes:
            - ReadWriteMany
   ```

1. Install helm chart

   ```bash
   helm install nextcloud stable/nextcloud -f nextcloud.values.yml -n nextcloud
   ```

**Expected Result**

No errors

**Actual Result** 

```bash
0s          Warning   Failed                   pod/nextcloud-mariadb-master-0                             Error: failed to start container "mariadb": Error response from daemon: error while creating mount source path '/var/lib/kubelet/pods/733aabd1-ed37-42a6-beb2-3e29ee4c24d5/volumes/kubernetes.io~csi/pvc-e280b38a-edd4-495e-8bd5-09389bf58238/mount': mkdir /var/lib/kubelet/pods/733aabd1-ed37-42a6-beb2-3e29ee4c24d5/volumes/kubernetes.io~csi/pvc-e280b38a-edd4-495e-8bd5-09389bf58238/mount: file exists
```

**Additional info**

```bash
# kubectl -n nextcloud get pvc  
NAME                              STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
data-nextcloud-mariadb-master-0   Bound    pvc-e280b38a-edd4-495e-8bd5-09389bf58238   8Gi        RWX            kadalu.replica1   11m
data-nextcloud-mariadb-slave-0    Bound    pvc-4c97e858-ab2d-44df-92e1-29d22c9c9fe6   8Gi        RWX            kadalu.replica1   11m
nextcloud-nextcloud               Bound    pvc-f999ed74-c682-41e1-b9bf-d067b84cee4d   500Gi      RWX            kadalu.replica1   11m
# kubectl -n nextcloud get pv
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS        CLAIM                                       STORAGECLASS      REASON   AGE
pvc-4c97e858-ab2d-44df-92e1-29d22c9c9fe6   8Gi        RWX            Delete           Bound         nextcloud/data-nextcloud-mariadb-slave-0    kadalu.replica1            11m
pvc-e280b38a-edd4-495e-8bd5-09389bf58238   8Gi        RWX            Delete           Bound         nextcloud/data-nextcloud-mariadb-master-0   kadalu.replica1            11m
pvc-f999ed74-c682-41e1-b9bf-d067b84cee4d   500Gi      RWX            Delete           Bound         nextcloud/nextcloud-nextcloud               kadalu.replica1            11m
```

**Comment by papanito** [comment]
Still valid I guess

**Comment by papanito** [comment]
Still valid I guess

**Comment by aravindavk** [comment]
> Still valid I guess

Thanks. I will try to reproduce this issue this week.

**Comment by amarts** [comment]
This was the same issue with kubelet changed in 1.20 k8s version... Now we perform makedirs(), previously the corresponding directories were tried to be created from kubelet itself, which errored out. Now its on CSI plugins, in which case its solved in 0.7.5 version.
